protobuf/src/test/scala/magnolify/protobuf/test/ProtobufTypeSpec.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package magnolify.protobuf.test
+
+import java.net.URI
+import java.time.Duration
+
+import cats._
+import cats.instances.all._
+import com.google.protobuf.{ByteString, Message}
+import magnolify.cats.auto._
+import magnolify.scalacheck.auto._
+import magnolify.protobuf._
+import magnolify.shims.JavaConverters._
+import magnolify.test.Proto2._
+import magnolify.test.Proto3._
+import magnolify.test.Simple._
+import magnolify.test._
+import org.scalacheck._
+
+import scala.reflect._
+
+object ProtobufTypeSpec extends MagnolifySpec("ProtobufRecordType") {
+  private def test[T: ClassTag: Arbitrary: Eq, U <: Message: ClassTag](
+    implicit tpe: ProtobufType[T, U],
+    eqt: Eq[T],
+    eqm: Eq[U] = Eq.instance[U](_.toByteString == _.toByteString)
+  ): Unit = {
+    ensureSerializable(tpe)
```

Comment:
Were you able to verify (just ad hoc) that Beam's `ProtoCoder` is able to encode/decode these? Or even just test that the generated Message classes can roundtrip using`Message#WriteTo` and `Message#parseFrom (since that's what ProtoCoder does)?

Reply:
Yeah it works. Proto ser/de is much simpler, if you can build it it's guaranteed to build.
Also `eqm` is copied from `AvroType` to deal with bad `GenericRecord` clone which should be impossible here. I'll remove.

But maybe we can add a ser/de check of avro/proto/example records for every `*Type` later.